roles/jd-server/src/lib/job_declarator/mod.rs

```diff
@@ -67,6 +70,24 @@ impl JobDeclaratorDownstream {
         }
     }
 
+    fn get_block_hex(self_mutex: Arc<Mutex<Self>>, message: SubmitSolutionJd) -> String {
+        //TODO: implement logic for success or error
+        let (last_declare, tx_list, _) = match self_mutex
+            .safe_lock(|x| x.declared_mining_job.take())
+            .unwrap()
+        {
+            Some((last_declare, tx_list, _x)) => (last_declare, tx_list, _x),
+            None => {
+                warn!("Received solution but no job available");
+                todo!()
```

Comment:
here we can just close the connection with downstream

Reply:
done